test/e2e/bgppolicy_test.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	"antrea.io/antrea/pkg/features"
+	"antrea.io/antrea/test/e2e/providers/exec"
+)
+
+type BGPPolicySpecBuilder struct {
+	Spec crdv1alpha1.BGPPolicySpec
+	Name string
+}
+
+var (
+	remoteASN = int32(65000)
+
+	localASN        = int32(64512)
+	updatedLocalASN = int32(64513)
+
+	password                   = "password"
+	defaultBGPPolicySecretName = "antrea-bgp-passwords" // #nosec G101
```

Comment:
why is there "Policy" in this variable name? I think the secret name is global and not specific to a policy?

Reply:
Updated to use the const `BGPPolicySecretName` in pkg/agent/types/bgppolicy.go